### ISSUE TITLE
ForccEmulatedFPS

### ISSUE THREAD

**Issue by Bassram32** [hrydgard/ppsspp#17152]
### What should happen

Removing ForceEmulatedFPS prevents me from playing many heavy games. 
that feature helped a lot in making hard to emulate games run smoothly, I could have never dreamed of playing Tomb Raider games if it wasn't for it.
Tekken 6 couldn't reach 60FPS but setting forced FPS to 50 did it without any kind of issues, same thing goes for WipeOut Pure and Pulse and God of War and Syphon Filter and Resistance Retribution. now I know setting Frameskip to 1 works as an alternative but why would I play at 30FPS while I can perfectly play at 50fps by the help of ForceEmulatedFPS?
Also, MotorSport Arctic Edge struggled to run on my device, but setting forced FPS to 27FPS made it perfect.

And no, it never caused in a game breaking bug or something of the sort.

I'm really sad that the hack is no longer available from 1.8.0 onward

### Who would this benefit

Weak hardware users

### Platform (if relevant)

None

### Games this would be useful in

many

### Checklist

- [X] Check the [latest git build](https://buildbot.orphis.net/ppsspp/) in case it's already implemented.
- [X] [Search for other requests](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same feature.

**Comment by LunaMoo** [comment]
ForceMaxFPS hack is messing with timing, it often results with okish sound speed, but gameplay/logic is actually slowed down, games ran at 50 fps with it aren't "working perfectly", they're running slower. Some games run certain screens at extremely high fps and with such hack those screens appear frozen and increase load times from a second or two to many minutes. That's just one problem why this hack was removed after reports of people thinking games were broken when they just were in a state of loading for a very long time.
 Well it wasn't really removed, just reduced to compat hack and only to it's safest 60 fps setting which previously was all that you had available from UI. If even it's safest option was degraded from UI to a compat setting, you should not expect it's more dangerous version to return.

 Why don't you just continue using older version that had the hack which made your experience "perfect". If you can't see the downsides of such hacks and have outdated hardware that can't benefit from cool new additions to emulator, why bother ever upgrading?

**Comment by ANR2ME** [comment]
As alternative, can't you turn off Frameskipping instead of setting it to 1? It will at least gives you a smooth movement even when it ran at only 50 fps.

**Comment by Gamemulatorer** [comment]
Duplicate of https://github.com/hrydgard/ppsspp/issues/11843

**Comment by Bassram32** [comment]
> 

of all the games I've tried, I can surely say that emulator renders every frame as intended and  doesn't slow down the games, except for two games, Phantasy Star Portable 2 Infinity and Assassin's Creed Bloodlines in which I noticed that the hack actually slows down the gameplay and it's very noticeable and you don't need a trained eye to tell if the game actually slowed down or runs at the given amount of frames assigned by the hack. those are two games out of 16 games I've tested with pleasing results

Old versions of PPSSPP have game breaking issues. example: Motorsports Arctic Edge not being able to go pass the first loading screen.

I understand the intention of not wanting the hack back, though

**Comment by Bassram32** [comment]
> ForceMaxFPS hack is messing with timing, it often results with okish sound speed, but gameplay/logic is actually slowed down, games ran at 50 fps with it aren't "working perfectly", they're running slower. Some games run certain screens at extremely high fps and with such hack those screens appear frozen and increase load times from a second or two to many minutes. That's just one problem why this hack was removed after reports of people thinking games were broken when they just were in a state of loading for a very long time. Well it wasn't really removed, just reduced to compat hack and only to it's safest 60 fps setting which previously was all that you had available from UI. If even it's safest option was degraded from UI to a compat setting, you should not expect it's more dangerous version to return.
> 
> Why don't you just continue using older version that had the hack which made your experience "perfect". If you can't see the downsides of such hacks and have outdated hardware that can't benefit from cool new additions to emulator, why bother ever upgrading?



> 

> 

of all the games I've tried, I can surely say that emulator renders every frame as intended and  doesn't slow down the games, except for two games, Phantasy Star Portable 2 Infinity and Assassin's Creed Bloodlines in which I noticed that the hack actually slows down the gameplay and it's very noticeable and you don't need a trained eye to tell if the game actually slowed down or runs at the given amount of frames assigned by the hack. those are two games out of 16 games I've tested with pleasing results

Old versions of PPSSPP have game breaking issues. example: Motorsports Arctic Edge not being able to go pass the first loading screen.

I understand the intention of not wanting the hack back, though

**Comment by Bassram32** [comment]
> As alternative, can't you turn off Frameskipping instead of setting it to 1? It will at least gives you a smooth movement even when it ran at only 50 fps.

No, if the game is forced to run at 60 it will always attempt to render all the frames per second. 50/60 framerates results in choppy sound and slow gameplay. 50/50 is smooth

**Comment by Gamemulatorer** [comment]
Try emulated cpu clock speed set to 111mhz

**Comment by hrydgard** [comment]
Now that we have a dedicated sections for "Speed hacks" that are clearly marked as possibly game breaking, things like this are a bit more reasonable. Still, the issue remains that this causes insane load times in a number of games that's not tiny, so some strong warning might be needed, or even more game specific flags, ugh.. 

At least this one is better than the old Timer Hack, though...
